langchain4j-open-ai/src/main/java/dev/langchain4j/model/openai/OpenAiImageModel.java

```diff
@@ -0,0 +1,149 @@
+     * @param model          dall-e-3 is default one
+     * @param persistTo      specifies the local path where the generated image will be downloaded to (in case provided).
+     *                       The URL within <code>dev.ai4j.openai4j.image.GenerateImagesResponse</code> will contain
+     *                       the URL to local images then.
+     * @param withPersisting generated response will be persisted under <code>java.io.tmpdir</code>.
+     *                       The URL within <code>dev.ai4j.openai4j.image.GenerateImagesResponse</code> will contain
+     *                       the URL to local images then.
+     */
+    @Builder
+    @SuppressWarnings("rawtypes")
+    public OpenAiImageModel(
+        @NonNull String apiKey,
+        String model,
+        Integer n,
+        String size,
+        String quality,
+        String style,
+        String user,
+        String responseFormat,
+        Duration timeout,
+        Integer maxRetries,
+        Proxy proxy,
+        Boolean logRequests,
+        Boolean logResponses,
+        Boolean withPersisting,
+        Path persistTo
+    ) {
+        timeout = getOrDefault(timeout, ofSeconds(60));
+
+        OpenAiClient.Builder cBuilder = OpenAiClient
+            .builder()
+            .openAiApiKey(apiKey)
+            .callTimeout(timeout)
+            .connectTimeout(timeout)
+            .readTimeout(timeout)
+            .writeTimeout(timeout)
+            .proxy(proxy)
+            .logRequests(getOrDefault(logRequests, false))
+            .logResponses(getOrDefault(logResponses, false))
+            .persistTo(persistTo);
+
+        if (withPersisting != null && withPersisting) {
+            cBuilder.withPersisting();
+        }
+
+        this.client = cBuilder.build();
+
+        this.maxRetries = getOrDefault(maxRetries, 3);
+        this.model = model;
+        this.n = n;
+        this.size = size;
+        this.quality = quality;
+        this.style = style;
+        this.user = user;
+        this.responseFormat = responseFormat;
+    }
+
+    public static class OpenAiImageModelBuilder {
+
+        public OpenAiImageModelBuilder withPersisting() {
+            withPersisting = true;
+            return this;
+        }
+
+        public OpenAiImageModelBuilder logRequests() {
+            logRequests = true;
+            return this;
+        }
+
+        public OpenAiImageModelBuilder logResponses() {
+            logResponses = true;
+            return this;
+        }
+    }
+
+    @Override
+    public Response<List<Image>> generate(String prompt) {
+        GenerateImagesRequest.Builder requestBuilder = GenerateImagesRequest
+            .builder()
+            .prompt(prompt)
+            .size(size)
+            .quality(quality)
+            .style(style)
+            .user(user)
+            .responseFormat(responseFormat);
+
+        if (DALL_E_2.equals(model)) {
+            requestBuilder.model(dev.ai4j.openai4j.image.ImageModel.DALL_E_2);
```

Comment:
minor: would be nice to have overloaded `model(String)` method in openai4j (as well as change `Builder.model` from `ImageModel` to `String`) to not limit only to these models in enum and not to force to make a conversion from string to enum here